packages/@aws-cdk/aws-logs-destinations/lib/kinesisfirehose.ts

```diff
@@ -0,0 +1,35 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as firehose from '@aws-cdk/aws-kinesisfirehose';
+import * as logs from '@aws-cdk/aws-logs';
+import { Stack } from '@aws-cdk/core';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from '@aws-cdk/core';
+
+/**
+ * Use a Kinesis Firehose as the destination for a log subscription
+ */
+export class KinesisFirehoseDestination implements logs.ILogSubscriptionDestination {
+  constructor(private readonly deliveryStream: firehose.DeliveryStream) {
+  }
+
+  public bind(_scope: Construct, _sourceLogGroup: logs.ILogGroup): logs.LogSubscriptionDestinationConfig {
+    const { region } = Stack.of(this.deliveryStream);
+
+    // Following example from https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/SubscriptionFilters.html#FirehoseExample
+    // Create a role to be assumed by CWL that can write to this Firehose.
+    const roleId = 'CloudWatchLogsCanPutRecordsIntoKinesisFirehose';
+    const role = this.deliveryStream.node.tryFindChild(roleId) as iam.IRole ??
+      new iam.Role(this.deliveryStream, roleId, {
```

Comment:
With your suggested change we would create 2 identical IAM roles that will fail my unit test in https://github.com/aws/aws-cdk/pull/14918/files#diff-4dab4ab34393d725389edf0c2c55b924e6e9c569d86f78b21ad5d36b8f29b076R104

Do we really want to create 2 identical roles in that unit test case?

Reply:
I think we do. There should be a logical mapping between the actor and the role, which would be violated if we create one role per delivery stream (as that is the resource). The test case can be modified to demonstrate that we re-use the role if there are multiple Firehose subscriptions from the same log group

Reply:
> The test case can be modified to demonstrate that we re-use the role if there are multiple Firehose subscriptions from the same log group

This would still create 2 separate roles because `scope` would be different for 2 separate subscriptions even if they are applied to the same log group.

So, it looks like `scope.node.tryFindChild(roleId) as iam.IRole ??` is a redundant check as that condition will never be true.

Anyway, I will just remove this unit test case for now as it's impossible to test that condition.

Reply:
Ah because the destination is bound under SubscriptionFilter, not LogGroup. Okay, no problem with having multiple roles then

Reply:
To be clear, yes we should remove the `tryFindChild` and always create a new role